### DISCUSSION TITLE
Generate controller error

### DISCUSSION THREAD

**Discussion by klippas-rob** [yippee-fun/phlex#235] [Help] [answered]
I'm new to Rails so this is likely my mistake…

`rails g controller cards`

produces:

```
      create  app/controllers/cards_controller.rb
       error  phlex [not found]
      invoke  test_unit
      create    test/controllers/cards_controller_test.rb
      invoke  helper
      create    app/helpers/cards_helper.rb
      invoke    test_unit
```

Prior to this I installed phlex with:

  ` bin/rails phlex:install`

and in my Gemfile I have:

  `gem "phlex", "~> 0.3.1"`

**Comment by joeldrapper** [comment]
Hey @klippas-rob, thanks for opening the issue. Super weird that it’s interfering with the controller generator. I wasn't able to reproduce this locally. Which version of Ruby and Rails are you using?

**Reply by klippas-rob** [reply]
Hi Joel,

ruby 3.1.2p20 
Rails 7.0.4

**Reply by klippas-rob** [reply]
Not sure if this is relevant but also using:
```
gem "simple_form"
gem "dry-validation"
gem "trailblazer"
gem "trailblazer-rails"
gem "reform-rails"
```

**Reply by joeldrapper** [reply]
Thanks. I’ve opened an issue #236 because this is probably a bug that needs to be fixed in Phlex.

**Reply by joeldrapper** [reply]
@klippas-rob do you think you could see if you can get this error in a fresh Rails app and send me steps you took from `rails new` to get there?

**Comment by klippas-rob** [comment]
It may be tomorrow, but will definitely get it to you.

**Comment by klippas-rob** [comment]
Hi Joel,

I found the problem.
It’s my fault—my apologies for creating this hassle for you.

I had this in the config/application.rb file:

	config.generators do |g|
 		 g.template_engine :phlex
	end

However, I’m a little puzzled.
When I remove the above code and do “rails generate controller …”, it invokes erb.
Is that expected?

	rails generate controller Cards
		 create app/controllers/cards_controller.rb 
		invoke erb
	…


Cheers,
Rob

**Reply by joeldrapper** [reply]
Ah, that makes sense. Yeah, Phlex is not a Rails template engine. It's a complete replacement for ActionView. I think we could find a way to hook into controller generation to have it generate Phlex views instead of ERB templates.

**Comment by klippas-rob** [comment, marked as answer]
Thanks Joel—again, apologies for the wild goose chase.
